### DISCUSSION TITLE
Alias support with gqlgen code

### DISCUSSION THREAD

**Discussion by chadgh** [99designs/gqlgen#2350] [Q&A]
I'm new to Go and gqlgen, but I wanted to know if there was native support to be able to submit multiple queries in a single request using the alias features in graphql. https://graphql.org/learn/queries/#aliases

So if I've got a schema like the following:

```graphql
type Book {
  tite: String!
  author: [Author!]!
}

type Author {
  name: String!
}
```

Could I define and submit a query like the following using what gqlgen provides? (Or how would I do the following using gqlgen? Is it even possible?)

```graphql
query ThreeFavoriteBooks {
  one: book(id: 5) {
    title
    author {
      name
    }
  }
  two: book(id: 7) {
    title
    author {
      name
    }
  }
  three: book(id: 14) {
    title
    author {
      name
    }
  }
}
```

**Comment by wilhelmeek** [comment]
@chadgh — assuming that your schema also includes
```gql
type Query {
 book(id: ID!): Book
}
```

And you've implemented all the generated resolvers, then your query will work out of the box.
